Zend/zend_hash.h

```diff
@@ -823,6 +825,27 @@ static zend_always_inline void *zend_hash_get_current_data_ptr_ex(HashTable *ht,
 		} \
 	} while (0)
 
+#define ZEND_HASH_FOREACH_END_DEL() \
+			__ht->nNumOfElements--; \
+			do { \
+				uint32_t j = HT_IDX_TO_HASH(_idx - 1); \
+				uint32_t nIndex = _p->h | __ht->nTableMask; \
+				uint32_t i = HT_HASH(__ht, nIndex); \
+				if (j != i) { \
+					Bucket *prev = HT_HASH_TO_BUCKET(__ht, i); \
+					while (Z_NEXT(prev->val) != j) { \
+						i = Z_NEXT(prev->val); \
+						prev = HT_HASH_TO_BUCKET(__ht, i); \
+					} \
+					Z_NEXT(prev->val) = Z_NEXT(_p->val); \
+				} else { \
+					 HT_HASH(__ht, _p->h | __ht->nTableMask) = Z_NEXT(_p->val); \
```

Comment:
Also could write this as `HT_HASH(__ht, nIndex) = Z_NEXT(_p->val)`, we already have the var.